extensions/cornerstone-dicom-rt/src/loadRTStruct.js

```diff
@@ -0,0 +1,217 @@
+import dcmjs from 'dcmjs';
+const { DicomMessage, DicomMetaDictionary } = dcmjs.data;
+const dicomlab2RGB = dcmjs.data.Colors.dicomlab2RGB;
+
+export default async function loadRTStruct(
+  extensionManager,
+  rtStructDisplaySet,
+  referencedDisplaySet,
+  headers
+) {
+  const utilityModule = extensionManager.getModuleEntry(
+    '@ohif/extension-cornerstone.utilityModule.common'
+  );
+
+  const { dicomLoaderService } = utilityModule.exports;
+  const imageIdSopInstanceUidPairs = _getImageIdSopInstanceUidPairsForDisplaySet(
+    referencedDisplaySet
+  );
+
+  // Set here is loading is asynchronous.
+  // If this function throws its set back to false.
+  rtStructDisplaySet.isLoaded = true;
+
+  const segArrayBuffer = await dicomLoaderService.findDicomDataPromise(
+    rtStructDisplaySet,
+    null,
+    headers
+  );
+
+  const dicomData = DicomMessage.readFile(segArrayBuffer);
+  const rtStructDataset = DicomMetaDictionary.naturalizeDataset(dicomData.dict);
```

Comment:
This is an expensive operation given that we have done it already on the main dataset, and we already have references.  Yes, there are times we need to do it, but we should avoid HAVING to refetch all the data if we don't have to.